pytorch_lightning/utilities/parsing.py

```diff
@@ -189,29 +190,25 @@ def lightning_hasattr(model, attribute):
         else:
             attr = hasattr(model.hparams, attribute)
     # Check if the attribute in datamodule (datamodule gets registered in Trainer)
-    elif trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
-        attr = getattr(trainer.datamodule, attribute)
-    else:
-        attr = False
+    if not attr and trainer is not None:
```

Comment:
what if the Trainer does not have `trainer.datamodule`?
UPDATE: I found the assignment in data connector, but missing in Trainer `__init__`
cc: @tchaton